yarn-project/types/src/create_tx_hash.ts

```diff
@@ -35,5 +37,17 @@ export function createTxHash({ newCommitments, newNullifiers, newContracts }: Tx
       newContracts.map(x => x.toBuffer()),
     ].flat(),
   );
-  return new TxHash(keccak(data));
+  return new TxHash(keccak224(data));
```

Comment:
Shouldn't we be just returning the first nullifier here?

Reply:
No, this is where we are computing the tx hash, i.e. what is going to go into the first nullifier slot. Eventually this will be done elsewhere (private kernel circuit perhaps), but for now it's done here.

Reply:
> what is going to go into the first nullifier slot

Then shouldn't this just depend on the `TxRequest` instead? Sorry, I think I may be missing something.

Reply:
There is no `SignedTxRequest` for private txs is there? Every tx now has it's hash inserted into the first nullifier slot. But how we compute the hash (in the sequencer for now) differs between private and public calls. Eventually it should be the same and done on the client. But for now we have this.